core/src/ast/mod.rs

```diff
@@ -206,9 +207,24 @@ impl ToSql for Statement {
                 source,
                 engine,
                 foreign_keys,
+                primary_key,
                 comment,
             } => {
                 let if_not_exists = if_not_exists.then_some("IF NOT EXISTS");
+
+                let primary_key = match (primary_key.as_ref(), columns.as_ref()) {
+                    (Some(indices), Some(columns)) => Some(format!(
+                        "PRIMARY KEY ({})",
+                        indices
+                            .iter()
+                            .copied()
+                            .map(|i| columns[i].name.clone())
+                            .collect::<Vec<_>>()
+                            .join(", ")
```

Comment:
nit. clone uses here are not necessary.
```rust
columns
    .iter()
    .enumerate()
    .filter_map(|(i, column)| {
        indices.contains(&i).then_some(column.name.as_str())
    })
    .collect::<Vec<_>>()
    .join(", ")
```

Reply:
Clone removed.

Any reason for switching from accessing by index to the contains? At this point in the code, we are guaranteed to have already validated the compatibility of columns and primary key indices, so if there is any out-of-bound access it would be relative to bugs in the code, and we would not want those to go unnoticed, right? A filter map may hide those.

Reply:
Resolved in commit #527f454e25f1e22146a96c762c3cff3b9870b068. Do let me know whether my approach is okay.

Reply:
both indices first and columns first, they don't handle the unreachable validated error case of out-of-bound access.
only different is that the first one panics, the other does not panic.
if we dont have a plan to handle error here, i thought it would be better to remove panicable code.

Reply:
While I agree that generally code that may panic is to be removed, here we are operating on an instantiated `CreateTable`, which carries several assumptions, one of which is that the primary key is compatible with the columns. If a non-self-consistent CreateTable were to be created, there should be a panic, as it is not an error relative to how the user is employing the library, but a library bug of which the panic would be an important signal.

Reply:
at least in the gluesql core workspace, we don't have plan to leave any kind of panicable codes. if you find it, that can be a bug and may be something that I missed during the review. we can fix it.
yes that can be some panicable case can happen. then, we can mark it using `Unreachable-` prefixed error rather than killing the program.
that has similar role but the only difference is that our library will not kill the application by itself.
of course we cannot make ideal, may be some dependent library has lots of unsafe codes and panicable codes, but at least we can control the code that we write.

Reply:
I have refactored the code into the version you initially proposed in commit 14f1f7a1504a50fa766e55fc4919ff621f6db45c